test/e2e/cluster_upgrade_test.go

```diff
@@ -0,0 +1,39 @@
+// +build e2e
+
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo"
+)
+
+var _ = Describe("When upgrading a workload cluster and testing K8S conformance [K8SVersion-Upgrade]", func() {
```

Comment:
I would suggest to document labels (`[K8SVersion-Upgrade]`) in https://cluster-api.sigs.k8s.io/developer/testing.html#running-the-end-to-end-tests and cross link the list of labels in https://cluster-api.sigs.k8s.io/developer/e2e.html

Also, are you planning to sunset the [Periodic-K8SVersion] label?

Reply:
PTAL